server/alerts_email_via_python3.py

```diff
@@ -0,0 +1,273 @@
+			description='Set connection to smtp server to use ssl/tls connection',
+			default=True
+		),
+		plugin_opts.OptionString(
+			name='email_jinja_template',
+			description='Password associated with smtp email',
+			default=''
+		),
+		plugin_opts.OptionString(
+			name='mail_options',
+			description='Additional mail options',
+			default=''
+		),
+	]
+	req_min_version = '1.3'
+	req_min_py_version = '3.0'
+	req_packages = {
+		'smtplib': has_smtp
+	}
+
+	def initialize(self):
+		signals.campaign_alert.connect(self.on_campaign_alert)
+		return True
+
+	def on_campaign_alert(self, table, alert_subscription, count):
+		user = alert_subscription.user
+		if not self.config['smtp_email']:
+			self.logger.debug("user {0} has no email address specified, skipping SMTP alert".format(user.id))
+			return False
+		if self.send_message(table, alert_subscription, count, self.config['smtp_email']):
+			return True
+		return False
+
+	def get_template_vars(self, table, alert_subscription, count):
+		campaign = alert_subscription.campaign
+		template_vars = {
+			'id': str(campaign.id),
+			'name': campaign.name,
+			'created': campaign.created,
+			'expiration': campaign.expiration,
+			'has_expired': campaign.has_expired,
+			'message_count': len(campaign.messages),
+			'visit_count': len(campaign.visits),
+			'credential_count': len(campaign.credentials),
+			'time': datetime.datetime.utcnow()
+		}
+		return template_vars
+
+	def create_headers(self, table, alert_subscription, count):
+
+		message = MIMEMultipart()
+		message['Subject'] = "Campaign Event: {0}".format(alert_subscription.campaign.name)
+		message['From'] = "<{0}>".format(self.config['smtp_email'])
+		message['To'] = "<{0}>".format(self.config['smtp_email'])
+
+		textual_message = MIMEMultipart('alternative')
+		txt_content = "{0:,} {1} reached for campaign: {2}".format(count, table.replace('_', ' '), alert_subscription.campaign.name)
+		plaintext_part = MIMEText(txt_content, 'plain')
+		textual_message.attach(plaintext_part)
+
+		email_jinja_template = ''
+		if os.path.isfile(self.config['email_jinja_template']):
+			with open(self.config['email_jinja_template'], 'r') as file_:
+				email_jinja_template = file_.read()
+
+		email_template = email_jinja_template if email_jinja_template else HTML_EMAIL_TEMPLATE
+		try:
+			rendered_email = Environment().from_string(email_template).render(template_vars=self.get_template_vars(table, alert_subscription, count))
+		except:
+			self.logger.warning('failed to render email jinja template', exc_info=True)
+			return False
+		html_part = MIMEText(rendered_email, 'html')
+		textual_message.attach(html_part)
+
+		message.attach(textual_message)
+		encoded_email = message.as_string()
+		return encoded_email
+
+	def send_message(self, table, alert_subscription, count, target_email):
+		msg = self.create_headers(table, alert_subscription, count)
+		if not msg:
+			return False
+		source_email = self.config['smtp_email']
+
+
+		if self.config['ssl_tls']:
```

Comment:
This block doesn't make sense. If `ssl_tls` is `False` then `server` is never set. `server` should be set with `SMTP` or `SMTP_SSL` depending on the configuration. If `ssl_tls` is `False` then `starttls` should be used as appliable.